src/OrchardCore.Modules/OrchardCore.Users/Views/RegistrationSettings.Edit.cshtml

```diff
@@ -1,27 +1,159 @@
+    <textarea asp-for="GenerateUsernameScript" rows="1" class="form-control content-preview-text"></textarea>
+    <span asp-validation-for="GenerateUsernameScript"></span>
+</div>
+
+<h5 class="form-group">@T["External Authentication"]<span class="hint"> — @T["Settings when registering with external authentication providers"]</span></h5>
+
+<div class="form-group" asp-validation-class-for="NoUsernameForExternalUsers">
+    <div class="custom-control custom-checkbox">
+        <input type="checkbox" class="custom-control-input" asp-for="NoUsernameForExternalUsers" data-external-registration />
+        <span asp-validation-for="NoUsernameForExternalUsers"></span>
+        <label class="custom-control-label" asp-for="NoUsernameForExternalUsers">@T["Generate username for external users"]</label>
+        <span class="hint">— @T["When a new user logs in with an external provider, he is not required to provide a local username. If GenerateUsername script is set, it is used to create the username."]</span>
+    </div>
+</div>
+<div class="form-group" asp-validation-class-for="NoEmailForExternalUsers">
+    <div class="custom-control custom-checkbox">
+        <input type="checkbox" class="custom-control-input" asp-for="NoEmailForExternalUsers" data-external-registration />
+        <span asp-validation-for="NoEmailForExternalUsers"></span>
+        <label class="custom-control-label" asp-for="NoEmailForExternalUsers">@T["Use email address from external provider"]</label>
+        <span class="hint">— @T["When a new user logs in with an external provider, he is not required to provide a local email address."]</span>
+    </div>
+</div>
+<div class="form-group" asp-validation-class-for="NoPasswordForExternalUsers">
+    <div class="custom-control custom-checkbox">
+        <input type="checkbox" class="custom-control-input" asp-for="NoPasswordForExternalUsers" data-external-registration />
+        <span asp-validation-for="NoPasswordForExternalUsers"></span>
+        <label class="custom-control-label" asp-for="NoPasswordForExternalUsers">@T["Do not create local password for external users"]</label>
+        <span class="hint">— @T["When a new user logs in with an external provider, he is not required to provide a local password."]</span>
+    </div>
+</div>
+
+<script at="Foot">
+    //<![CDATA[
+    function refresh(e) {
+        var usersCanRegisterElement = document.getElementById("@Html.IdFor(m => m.UsersCanRegister)");
+        var externalRegistrationInputs = $("input[data-external-registration]");
+
+        if (usersCanRegisterElement.value == "NoRegistration") {
+            externalRegistrationInputs.prop('disabled', true);
+        } else {
+            externalRegistrationInputs.prop('disabled', false);
+        }
+    }
+
+    function resetScript(keepText) {
+        var editor = $('#@Html.IdFor(x => x.GenerateUsernameScript)').data('editor');
+        if (!keepText) {
+            editor.doc.setValue(
```

Comment:
I don't understand what this is used for!!

Reply:
When there is no script saved in the settings, it initializes the default script on load. 
@hishamco thanx for the review.